templates/.github/workflows/ci.yml

```diff
@@ -0,0 +1,153 @@
+name: CI Pipeline
+on:
+  push:
+    branches: [master, main]
+env:
+  # Environment variables shared across jobs and doesn't change per environment
+  region: <% index .Params `region` %>
+  accountId: "<% index .Params `accountId` %>"
+  repo: <% .Name %>
+  namespace: <% .Name %>
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+    env:
+      CI: true
+      build_env: production
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run Unit Tests
+      uses: actions/setup-go@v2
+    - run: |
+        go get -u github.com/jstemmer/go-junit-report
+        mkdir -p test-reports
+        go test -v ./...
+
+  build:
+      needs: unit-test
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+        - name: Set IMAGE_TAG as env
+          run: |
+            IMAGE_TAG=$(git rev-parse --short=7 ${{ github.sha }})
+            echo "IMAGE_TAG=${IMAGE_TAG}" >> $GITHUB_ENV
+        - name: Configure AWS credentials
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+            aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+            aws-region: ${{ env.region }}
+        - name: Login to Amazon ECR
+          id: login-ecr
+          uses: aws-actions/amazon-ecr-login@v1
+        - name: Build and push
+          id: docker_build
+          uses: docker/build-push-action@v2
+          with:
+            push: true
+            tags: ${{ env.accountId }}.dkr.ecr.${{ env.region }}.amazonaws.com/${{ env.repo }}:${{ env.IMAGE_TAG }},${{ env.accountId }}.dkr.ecr.${{ env.region }}.amazonaws.com/${{ env.repo }}:latest
+
+  deploy:
```

Comment:
Should this be called `staging-deploy` since the later one is `production-deploy`?